spec/controllers/activity_forms_controller_spec.rb

```diff
@@ -346,6 +346,14 @@
       end
     end
 
+    context "when setting non-ODA on a programme" do
+      it "updates the RODA identifier to start with 'NODA'" do
+        put :update, params: {activity_id: programme.id, id: :is_oda, activity: {is_oda: false}}
```

Comment:
You could use the `put_step` method here

Reply:
I could, but then I'd have to recreate the already defined `activity` to be a programme instead of using the `programme`. (`put_step` is hardcoded to use `activity.id`).

Reply:
Would `let(:activity) { programme }` in the context do the job? Although maybe it would be unclear what the purpose of that is when it's abstracted away in `put_step`